### PR TITLE
[PF 22]: Add security schema for Janitor API

### PR DESCRIPTION
**Pull request by yonghaoy**
Part of https://broadworkbench.atlassian.net/browse/PF-22

### CODE REVIEW

File: terra-resource-janitor-client/build.gradle [hunk 1]
```diff
@@ -20,6 +20,8 @@ dependencies {
     implementation group: 'org.glassfish.jersey.core', name: 'jersey-client', version: "${jersey}"
     implementation group: 'org.glassfish.jersey.media', name: 'jersey-media-json-jackson', version: "${jersey}"
     implementation group: 'org.glassfish.jersey.media', name: 'jersey-media-multipart', version: "${jersey}"
+    implementation "org.glassfish.jersey.inject:jersey-hk2:${jersey}"
```
Comment (wchamber) [inline terra-resource-janitor-client/build.gradle:23]: what do we need this for?

File: src/main/resources/api/service_openapi.yaml [hunk 2]
```diff
@@ -186,3 +193,17 @@ components:
         application/json:
           schema:
             $ref: '#/components/schemas/ResourceDescription'
+
+  securitySchemes:
```
Comment (wchamber) [inline src/main/resources/api/service_openapi.yaml:197]: do we need to add `security[something]` in our yaml somewhere?

Comment (yonghaoy) [reply src/main/resources/api/service_openapi.yaml:197]: Compared with SAM, Data Repo and WM code, seems they are similar
I think we won't need to add `security: []` for unthenticatevd APIs. For others, and seems we don't need that

File: terra-resource-janitor-client/build.gradle (hunk 1 above)
Comment (yonghaoy) [reply terra-resource-janitor-client/build.gradle]: Ah, I think I can remove this one. I got some jersey binding exceptions in CRL when uses janitor calls. I saw data repo had that and it works. But I also added the same dependency in CRL, and it solves that problem. Let me delete this. 
Same issue as this: https://stackoverflow.com/questions/44088493/jersey-stopped-working-with-injectionmanagerfactory-not-found

**Pull request merged**